lib/spack/spack/build_systems/python.py

```diff
@@ -97,10 +97,11 @@ def configure(self, spec, prefix):
 
     extends('python')
 
-    def setup_file(self, spec, prefix):
+    def setup_file(self):
```

Comment:
Why this particular change? This function is only needed for a single file, and that package needs to access the spec.

We might want to save things that change between self and self/spec/prefix for another PR. You, @tgamblin, and I all disagree on the best way to handle this 😆